### DISCUSSION TITLE
Need help getting raw line to work

### DISCUSSION THREAD

**Discussion by lettuceDestroyer** [typst/typst#3958] [Q&A] [answered]
I can't get `raw.line()` to work.

What I tried is the following:

Typst file:
```typst
#set raw(theme: "dark_modern.tmTheme")
#show raw: it => block(
  fill: rgb("#1d2433"),
  inset: 8pt,
  radius: 5pt,
  text(fill: rgb("#a2aabc"), it)
)

#raw.line(1,3,"System",
  raw(read("./CSharp_Example_File.cs"), lang: "cs")
)
````

`CSharp_Example_File.cs`:
```cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProbeIPA.Shared.DTO
{
    public class BuildingDto
    {
        public int Id { get; set; }
        public List<RoomDto> Rooms { get; set; }
        public List<UserDto> FacilityManagers { get; set; }
        public string Test { get; set; }
    }

    public void Test()
    {
        string name = "\"Tobi\"";
    }
}
```

The tmTheme can be found in [this repository](https://github.com/lettuceDestroyer/vscode-dark-modern-tmtheme.git)

Additional information:
- Editor: VSCode v1.88.1 with Extensions
    - _Typst LSP v0.13.0_
    - _Typst Preview v0.11.4_

**Comment by IllustratedMan-code** [comment, marked as answer]
What exactly are you trying to do with `raw.line`? Are you trying to highlight the "system" text in the first 3 lines? 

You can't use `raw.line` that way, though you can modify it with show rules. I managed to highlight the first three "System" occurrences using the following code:
 ```typst
#{
  show raw.line: it => if it.number <= 3 {
    show "System": set text(blue)
    it
  } else {
    it
  }
raw(
  read("./CSharp_Example_File.cs"),
  lang: "cs",
  )
}
```


![image](https://github.com/typst/typst/assets/60514384/079eadda-6a56-4978-bbf4-ed80bde7d70b)
